File: is_prime/turingarena-files/C/skeleton.c
#include <stdio.h>
#include <stdlib.h>


int is_prime(int n);


int main() {
    // checkpoint
    printf("%d\n", 0);
    // read n
    static int n;
    fflush(stdout);
    scanf("%d", &n);
    // call prime = is_prime(n)
    static int prime;
    prime = is_prime(n);
    // write prime
    printf("%d\n", prime);
    // exit
    exit(0);
}

File: is_prime/turingarena-files/Rust/skeleton.rs
mod solution;
use std::io::Write;

macro_rules! readln {
    ($($var:expr),*) => {{
        let mut buf = String::new();
        std::io::stdin().read_line(&mut buf).unwrap();
        let parts: Vec<&str> = buf.trim().split(" ").collect();
        let mut i: usize = 0;
        $(
            assert!(i < parts.len(), "input format incorrect: too few values on this line"); 
            $var = parts[i].parse().unwrap();
            i += 1;
        )*
        assert!(i == parts.len(), "input format incorrect: too many values on the line")
    }};
}


fn main() {
    // checkpoint
    println!("{}", 0);
    // read n
    let n: i64;
    std::io::stdout().flush().unwrap();
    readln!(n);
    // call prime = is_prime(n)
    let prime: i64;
    prime = solution::is_prime(n);
    // write prime
    println!("{}", prime);
    // exit
    std::process::exit(0);
}
